client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -155,6 +156,15 @@
     */
    NotifyingFuture<Boolean> replaceWithVersionAsync(K key, V newValue, long version, int lifespanSeconds, int maxIdleSeconds);
 
+   /**
+    * Retrieve entries from the server
+    *
+    * @param filterConverterFactory Factory name for the KeyValueFilterConverter or null
+    * @param segments               The segments to iterate on or null if all segments should be iterated
+    * @param batchSize              The number of entries transferred from the server at a time
+    * @return Iterator for the entries
+    */
+   CloseableIterator<Entry<Object, Object>> retrieveEntries(String filterConverterFactory, Set<Integer> segments, int batchSize);
```

Comment:
@gustavonalle Note that I'm planning to make the mapping of keys to segments configurable: https://issues.jboss.org/browse/ISPN-5465. I was also thinking of calling it `Partitioner`, but then I thought there may be confusion with the partition handling functionality.